### DISCUSSION TITLE
Testing circular imports using `pytest`

### DISCUSSION THREAD

**Discussion by kloczek** [pytest-dev/pytest#12124] [General] [answered]
I'm looking for some way of testing module code against circular imports.
Someone can recommend how to do that using pytest?🤔

**Comment by nicoddemus** [comment]
Probably pytest is not a good fit for this, because it imports modules to analyse them, which will break if there are circular imports. 

It is probably best done by a tool which analyses the files without importing.

**Reply by Pierre-Sassoulas** [reply, marked as answer]
Pylint can do that with `--enable=cyclic-import` and there's a pylint-pytest plugin if you really want to have it included in pytest.

**Reply by nicoddemus** [reply]
Thanks, I was sure something like that existed, but was out of time to search for it. 👍
